Description: Fix upgrade of an empty DB from 2.6.0 to 2.6.1
 Happens when you install schooltool on saucy, then add a ppa, and upgrade.
Origin: upstream, https://bazaar.launchpad.net/~schooltool-owners/schooltool/trunk/revision/3717
Bug: https://bugs.launchpad.net/bugs/1258529
Index: schooltool/src/schooltool/generations/evolve42.py
===================================================================
--- schooltool.orig/src/schooltool/generations/evolve42.py
+++ schooltool/src/schooltool/generations/evolve42.py
@@ -26,15 +26,15 @@ from zope.annotation.interfaces import I
 from zope.component.hooks import getSite, setSite
 
 from schooltool.app.interfaces import ISchoolToolApplication
+from schooltool.app.app import getApplicationPreferences
 
 SCHEDULES_KEY = 'schooltool.timetable.schedules'
 TIMETABLES_KEY = 'schooltool.timetable.timetables'
-APP_PREFERENCES_KEY = 'schooltool.app.ApplicationPreferences'
 
 
 def evolveScheduleContainers(app):
     annotations = IAnnotations(app)
-    prefs = annotations[APP_PREFERENCES_KEY]
+    prefs = getApplicationPreferences(app)
     schedule_containers = app[SCHEDULES_KEY]
     for container in schedule_containers.values():
         if container.timezone != prefs.timezone:
@@ -44,7 +44,7 @@ def evolveScheduleContainers(app):
 
 def evolveTimetableContainers(app):
     annotations = IAnnotations(app)
-    prefs = annotations[APP_PREFERENCES_KEY]
+    prefs = getApplicationPreferences(app)
     schedule_containers = app[TIMETABLES_KEY]
     for container in schedule_containers.values():
         for timetable in container.values():
